tests/dialects/test_stencil.py

```diff
@@ -40,3 +49,103 @@ def test_stencil_return_multiple_ResultType():
     assert return_op.arg[0] is result_type_val1
     assert return_op.arg[1] is result_type_val2
     assert return_op.arg[2] is result_type_val3
+
+
+def test_stencil_cast_op_verifier():
+    field = TestSSAValue(FieldType.from_shape((-1, -1, -1), f32))
+
+    # check that correct op verifies correctly
+    cast = CastOp.get(
+        field,
+        IndexAttr.get(-2, -2, -2),
+        IndexAttr.get(100, 100, 100),
+        FieldType.from_shape((102, 102, 102), f32),
+    )
+    cast.verify()
+
+    # check that math is correct
+    with pytest.raises(VerifyException):
+        cast = CastOp.get(
+            field,
+            IndexAttr.get(-2, -2, -2),
+            IndexAttr.get(100, 100, 100),
+            FieldType.from_shape((100, 100, 100), f32),
+        )
+        cast.verify()
```

Comment:
Don't you want to check which exceptions are thrown exactly? Or what are these tests supposed to do?

Reply:
How can I check what is thrown exactly? I think it'd be nice to check that!